d3-selection/index.d.ts

```diff
@@ -845,7 +878,7 @@ interface Selection<GElement extends BaseType, Datum, PElement extends BaseType,
     /**
      * Return the first (non-null) element in this selection. If the selection is empty, returns null.
      */
-    node(): GElement | null;
+    node(): GElement;
```

Comment:
I'm not crazy about this change. The reason is that it is super easy to accidentally lie to the compiler. Any time you pass in an element type to `d3.select` or `d3.selectAll`, you get a non-empty selection:

```typescript
const oops= d3.select<HTMLDivElement>('div.whoops').node();
console.log(oops.parentElement); // crash
```

I make heavy use of the type parameter argument, thanks to the SVG DOM -- sometimes you just gotta `.each()` and compute some metrics the old-fashioned way.

We can't guarantee that any selection is valid, so I would argue for keeping the return type `GElement | null` despite the seeming redundancy. Users who know that the node is valid (or are willing to accept the possibility of a crash at runtime) can use the postfix `!` operator:

```typescript
const whoops = d3.select<HTMLDivElement>('div.whoops').node();
console.log(whoops!.parentElement);
```